aws-rds-dbparametergroup/src/main/java/software/amazon/rds/dbparametergroup/BaseHandlerStd.java

```diff
@@ -73,6 +81,62 @@
+            final Tagging.TagSet desiredTags
+    ) {
+        final Tagging.TagSet tagsToAdd = Tagging.exclude(desiredTags, previousTags);
+        final Tagging.TagSet tagsToRemove = Tagging.exclude(previousTags, desiredTags);
+
+        if (tagsToAdd.isEmpty() && tagsToRemove.isEmpty()) {
+            return progress;
+        }
+
+        final String arn = progress.getCallbackContext().getDbParameterGroupArn();
+
+        try {
+            Tagging.removeTags(rdsProxyClient, arn, Tagging.translateTagsToSdk(tagsToRemove));
+            Tagging.addTags(rdsProxyClient, arn, Tagging.translateTagsToSdk(tagsToAdd));
+        } catch (Exception exception) {
+            return Commons.handleException(
+                    progress,
+                    exception,
+                    SOFT_FAIL_NPROGRESS_TAGGING_ERROR_RULE_SET
+            );
+        }
+
+        return progress;
+    }
+
+    protected CallChain.Completed<DescribeDbParameterGroupsRequest, DescribeDbParameterGroupsResponse, RdsClient, ResourceModel, CallbackContext>
+    describeDbParameterGroups(final AmazonWebServicesClientProxy proxy,
+                              final ProxyClient<RdsClient> proxyClient, final ResourceModel model,
+                              final CallbackContext callbackContext) {
+        return proxy.initiate("rds::read-db-parameter-group", proxyClient, model, callbackContext)
+                .translateToServiceRequest(Translator::describeDbParameterGroupsRequest)
+                .backoffDelay(CONSTANT)
+                .makeServiceCall((describeDbParameterGroupsRequest, proxyInvocation) -> proxyInvocation.injectCredentialsAndInvokeV2(describeDbParameterGroupsRequest, proxyInvocation.client()::describeDBParameterGroups))
+                .handleError((describeDbParameterGroupsRequest, exception, client, resourceModel, ctx) ->
+                        Commons.handleException(
+                                ProgressEvent.progress(resourceModel, ctx),
+                                exception,
+                                DEFAULT_DB_PARAMETER_GROUP_ERROR_RULE_SET
+                        ));
+    }
+
+    protected ProgressEvent<ResourceModel, CallbackContext> fetchDBParameterGroupArn(final AmazonWebServicesClientProxy proxy,
+                                                                                     final ProxyClient<RdsClient> proxyClient,
+                                                                                     final ProgressEvent<ResourceModel, CallbackContext> progress) {
+        return describeDbParameterGroups(proxy, proxyClient, progress.getResourceModel(), progress.getCallbackContext())
+                .done((describeDbParameterGroupsRequest, describeDbParameterGroupsResponse, invocation, resourceModel, context) -> {
+                    final String arn = describeDbParameterGroupsResponse.dbParameterGroups().stream().findFirst().get().dbParameterGroupArn();
+                    context.setDbParameterGroupArn(arn);
```

Comment:
I recall your comment about this approach in DBInstance. This is exactly what I pictured it back then. I still haven't decided if I find the context population approach any reliable, let's A/B test it.